plugins/scenario/hu.bme.mit.gamma.scenario.statechart.util/src/hu/bme/mit/gamma/scenario/statechart/util/ScenarioStatechartUtil.java

```diff
@@ -129,13 +146,41 @@ public String getHotComponentViolation() {
 	public String getHotEnvironmentViolation() {
 		return hotEnvironmentViolation;
 	}
-	
+
 	public String getCombinedStateAcceptingName(String name) {
 		return name + "__" + Accepting;
 	}
-	
+
 	public String getNameOfNewPort(Port port, boolean isSend) {
 		return isSend ? getTurnedOutPortName(port) : port.getName();
 	}
 
+	public List<EventReference> getAllEventReferencesForDirection(Component automaton, boolean isSentByComponent) {
+		List<EventReference> eventRefs = new LinkedList<>();
+		List<Port> correctPorts = automaton.getPorts().stream()
+				.filter((it) -> (!((StatechartModelDerivedFeatures.getInputEvents(it)).isEmpty())))
+				.collect(Collectors.toList());
+		for (Port port : correctPorts) {
+			if ((isTurnedOut(port) && isSentByComponent) || (!isTurnedOut(port) && !isSentByComponent)) {
+				for (EventDeclaration eventDecl : port.getInterfaceRealization().getInterface().getEvents()) {
```

Comment:
Use `StatechartModelDerivedFeatures.getAllEvents()` to retrieve all events on a port.